src/config/config.rs

```diff
@@ -113,6 +113,31 @@ impl Config {
     pub fn contains(&self, name: &str) -> bool {
         self.types.contains_key(name) || self.unions.contains_key(name)
     }
+
+    pub fn get_all_used_type_names(&self) -> HashSet<String> {
+        let mut set = HashSet::new();
+        let mut stack = vec!["Query".into(), "Mutation".into()];
+        while let Some(type_name) = stack.pop() {
+            if let Some(typ) = self.types.get(&type_name) {
+                set.insert(type_name);
+                for field in typ.fields.values() {
+                    stack.extend(field.args.values().map(|arg| arg.type_of.clone()));
+                    stack.push(field.type_of.clone());
+                }
+            }
+        }
+
+        set
+    }
+
+    pub fn remove_unused_types(&mut self) {
+        let used_types = self.get_all_used_type_names();
+        let all_types: HashSet<String> = self.types.keys().cloned().collect();
+        let unused_types = all_types.difference(&used_types);
+        for unused_type in unused_types {
+            self.types.remove(unused_type);
```

Comment:
add another method to remove a type from config `fn remove_type(self, name: &str) -> Self`

Reply:
add another method `fn unused_types(&self) -> HashSet<String>`

Reply:
Make sure to add meaningful docs on each method.